transformer_engine/pytorch/ops/sequential.py

```diff
@@ -39,7 +39,7 @@ def __init__(
         self._module_groups = None
 
         # Add modules
-        if len(args) == 1 and isinstance(args[0], OrderedDict):
```

Comment:
Wouldn't this then tie us to this new version of pyTorch? We should probably handle both cases, right?

Reply:
[`collections.OrderedDict`](https://docs.python.org/3/library/collections.html#collections.OrderedDict) is a subclass of `dict`, so this change is backward-compatible. `dict`s preserve insertion order since Python 3.6 (see [comments here](https://docs.python.org/3/library/stdtypes.html#mapping-types-dict)), so there's not much harm in relaxing the constraint.

`dict` inputs are a weird case that we mainly support because it's supported in `torch.nn.Sequential`:
https://github.com/pytorch/pytorch/blob/cd8bbdc71a0258292381a7d54c8b353988d02ff4/torch/nn/modules/container.py#L121-L123. I'm tempted to remove our reliance on this API. Ironically `torch.nn.Sequential` doesn't seem to use this API internally either.